libosmscout-client-qt/src/osmscoutclientqt/TiledMapRenderer.cpp

```diff
@@ -418,10 +418,17 @@ void TiledMapRenderer::onLoadJobFinished(QMap<QString,QMap<osmscout::TileKey,osm
             (double)loadXFrom + (double)width/2.0,
             (double)loadYFrom + (double)height/2.0);
 
-    double osmTileDimension = (double)OSMTile::osmTileOriginalWidth() * (mapDpi / OSMTile::tileDPI() ); // pixels
+    /** tiles are up-scaled almost two times, when zoom level is near its next integer value (for example 16.99)
+     * to avoid blured result, we render tiles two times bigger than its minimum visual size.
+     * Another possible upscale (screenPixelRatio) is for HiDPI screens - when there is ratio 2.0, 100px on Qt canvas
+     * is displayed as 200px on the screen. To provide best results on HiDPI screen, we upscale tiles by this pixel ratio.
+     */
+    double finalDpi = mapDpi * 2 * this->screenPixelRatio;
```

Comment:
I am not sure if we should not have this basic upscale configurable. Two is pretty huge performance impact :-/